### PR TITLE
Remove deprecated `--check-cfg` syntax

### PR DESCRIPTION
**Pull request by Urgau**
This PR removes the deprecated `--check-cfg` `names(...)` and `values(...)` syntax.

Follow up to https://github.com/rust-lang/rust/pull/111072
Part of https://github.com/rust-lang/compiler-team/issues/636

r? compiler

### CODE REVIEW

File: src/tools/rust-analyzer/Cargo.lock [hunk 1]
```diff
@@ -888,9 +888,9 @@ checksum = "e2abad23fbc42b3700f2f279844dc832adb2b2eb069b2df918f455c4e18cc646"
 
 [[package]]
 name = "libc"
-version = "0.2.148"
+version = "0.2.150"
```
Comment (lnicola) [inline src/tools/rust-analyzer/Cargo.lock:891]: Is this relevant and necessary?

**Comment by Urgau** [comment]
> Some changes occurred in src/tools/rust-analyzer

I've update the `libc` version to 0.2.150 which includes some changes necessary for this PR. Let me know if you prefer those changes to be directly made in the rust-analyzer repo.

**Comment by lnicola** [comment]
> Let me know if you prefer those changes to be directly made in the rust-analyzer repo.

Yes, very much, but if it's required for this PR, we'll need a sync PR before this lands.

File: src/tools/rust-analyzer/Cargo.lock (hunk 1 above)
Comment (Urgau) [reply src/tools/rust-analyzer/Cargo.lock:891]: *(wow you are too fast, didn't have time to fully write the comment explaining this change)*

Yes, it includes https://github.com/rust-lang/libc/pull/3410 which was published in 0.2.150

**Review by b-naber — APPROVED** [review]

**Comment by fee1-dead** [comment]
https://github.com/rust-lang/rust/pull/118641#issuecomment-1840661295

@bors r-

**Comment by Urgau** [comment]
After some trouble finding a runner that actually test the step that broke, I can confirm that it is solved by https://github.com/rust-lang/rust/pull/117981/commits/801bc561bc9c2ff73ce9c15c6636e6dd5d060bf3 (cf. temporary [x86_64-gnu](https://github.com/rust-lang/rust/actions/runs/7101274720/job/19329049713?pr=117981#step:26:5447) runner).

The PR can be re-approved.

@rustbot labels -T-bootstrap -T-infra -A-testsuite -S-waiting-on-author +S-waiting-on-review

**Pull request merged**